### PR TITLE
Expose event-type in stream interface

### PR DESCRIPTION
**Pull request by dagnir**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java [hunk 1]
```diff
@@ -203,6 +210,30 @@ private void addEventSupport(TypeSpec.Builder specBuilder, ShapeModel eventStrea
...
+
+        if (usesLegacyScheme) {
+            specBuilder.addMethod(MethodSpec.methodBuilder("sdkEventType")
+                       .addAnnotation(Override.class)
+                       .addModifiers(PUBLIC)
+                       .returns(String.class)
+                       .addStatement("return $S", eventName)
+                       .build());
+        }
+
```
Comment (zoewangg) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java:236]: nit: can we extract it to a separate method?

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/eventstream.java [hunk 2]
```diff
@@ -41,6 +41,13 @@ public void accept(StreamDeathsResponseHandler.Visitor visitor) {
         return DefaultPerson.builder();
     }
 
+    /**
+     * The type of this event. Corresponds to the {@code :event-type} header on the Message.
+     */
+    default String sdkEventType() {
```
Comment (zoewangg) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/eventstream.java:47]: Should we use enum for the type?

Comment (dagnir) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/eventstream.java:47]: What's the benefit?

Comment (zoewangg) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/eventstream.java:47]: It'll be easier to enumerate all types when they do `switch`, whereas with string, they will need to manually check the code to see all available types.

Comment (dagnir) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/eventstream.java:47]: Sounds good.

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaulteventtwo.java [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+import software.amazon.awssdk.services.json.model.EventStreamOperationResponseHandler;
+import software.amazon.awssdk.services.json.model.EventStreamOperationWithOnlyOutputResponseHandler;
+import software.amazon.awssdk.services.json.model.EventTwo;
 
 /**
- * A specialization of {@code software.amazon.awssdk.services.jsonprotocoltests.model.EventTwo} that represents the
- * {@code EventStream#EventTwo} event. Do not use this class directly. Instead, use the static builder methods on
- * {@link software.amazon.awssdk.services.jsonprotocoltests.model.EventStream}.
+ * A specialization of {@code software.amazon.awssdk.services.json.model.EventTwo} that represents the
+ * {@code EventStream$EventTwo} event. Do not use this class directly. Instead, use the static builder methods on
```
Comment (zoewangg) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaulteventtwo.java:12]: Typo? `EventStream$EventTwo` -> `EventStream#EventTwo`

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaultsecondeventtwo.java [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
- * {@code EventStream#SecondEventTwo} event. Do not use this class directly. Instead, use the static builder methods on
- * {@link software.amazon.awssdk.services.jsonprotocoltests.model.EventStream}.
+ * A specialization of {@code software.amazon.awssdk.services.json.model.EventTwo} that represents the
+ * {@code EventStream$secondeventtwo} event. Do not use this class directly. Instead, use the static builder methods on
+ * {@link software.amazon.awssdk.services.json.model.EventStream}.
  */
 @SdkInternalApi
 @Generated("software.amazon.awssdk:codegen")
-public final class DefaultSecondEventTwo extends EventTwo {
+public final class DefaultSecondeventtwo extends EventTwo {
```
Comment (zoewangg) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaultsecondeventtwo.java:17]: Is this change intentional? Shouldn't it be `DefaultSecondEventTwo`?

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaulteventtwo.java (hunk 3 above)
Comment (dagnir) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaulteventtwo.java:12]: No, decided to use the C2j notation

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaultsecondeventtwo.java (hunk 4 above)
Comment (dagnir) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/defaultsecondeventtwo.java:17]: It's a little confusing, but this one corresponds to this event https://github.com/aws/aws-sdk-java-v2/blob/32cd8d61bc470ef325477fa29af90d5b8c12b2b4/codegen/src/test/resources/software/amazon/awssdk/codegen/poet/eventstream/service-2.json#L106

**Review by zoewangg — APPROVED** [review]
